web_client/components/containers/upload.jsx

```diff
@@ -5,7 +5,7 @@ import Upload from '../body/upload';
 import actions from '../../actions';
 import { rest } from '../../globals';
 import objectReduce from '../../utils/object-reduce';
-import { Promise } from '../../utils/promise';
+import { Promise, SCHEDULES, map as promiseMap } from '../../utils/promise';
```

Comment:
I agree that monkeypatching `Promise.map` is probably a bad idea.

I've been bitten in other projects by bad name clashes that weren't so easily fixable with renaming imports as in ES6. That's the main reason for my caution here. But if we want to support this workflow by the client of renaming things from the most accurate description to something that doesn't clash, that's ok with me.